### PR TITLE
CLD-497: Docker scripts should retry and timeout init and join steps

### PR DESCRIPTION
**Pull request by ilanRosenbaum**

### CODE REVIEW

File: src/centos/scripts/start-marklogic.sh [hunk 1]
```diff
@@ -96,6 +98,61 @@ if [ -n "${TZ}" ]; then
...
+################################################################
+# restart_check(hostname, baseline_timestamp, caller_lineno)
+#
+# Use the timestamp service to detect a server restart, given a
+# a baseline timestamp. Use N_RETRY and RETRY_INTERVAL to tune
+# the test length. Include authentication in the curl command
+# so the function works whether or not security is initialized.
+#   $1 :  The hostname to test against
+#   $2 :  The baseline timestamp
+#   $3 :  Invokers LINENO, for improved error reporting
```
Comment (sumanthravipati) [inline src/centos/scripts/start-marklogic.sh:110]: We are not using third parameter $3 anywhere. Description says for improved error reporting, but couldn't find where it is used in restart_check()

File: src/centos/scripts/start-marklogic.sh [hunk 2]
```diff
@@ -172,17 +234,23 @@ elif [[ "${MARKLOGIC_INIT}" == "true" ]]; then
     fi
 
     log "Initialzing MarkLogic on ${HOSTNAME}."
-
-    curl -s --anyauth -i -X POST \
-        -H "Content-type:application/json" \
+    TIMESTAMP=$(curl --anyauth -m 20 -s --retry 5 \
```
Comment (sumanthravipati) [inline src/centos/scripts/start-marklogic.sh:237]: Better to have timeout set to same value either 20 or 30. Not both is different places

Comment (ilanRosenbaum) [reply src/centos/scripts/start-marklogic.sh:237]: Agreed, I missed that when the change to 30s was made. Fixed now.

File: src/centos/scripts/start-marklogic.sh (hunk 1 above)
Comment (ilanRosenbaum) [reply src/centos/scripts/start-marklogic.sh:110]: I copied that directly from the docs, comment included, we are not using it and that comment should be removed

File: src/centos/scripts/start-marklogic.sh [hunk 3]
```diff
@@ -96,6 +98,60 @@ if [ -n "${TZ}" ]; then
     echo "${TZ}" | sudo tee /etc/timezone
 fi
 
+################################################################
+# restart_check(hostname, baseline_timestamp, caller_lineno)
```
Comment (sumanthravipati) [inline src/centos/scripts/start-marklogic.sh:102]: Please update this too with correct parameters count.

**Review by sumanthravipati — APPROVED** [review]

**Review by barkhachoithani — APPROVED** [review]

**Pull request merged**